rtmp-client/src/main/cpp/librtmp-jni.c

```diff
@@ -37,17 +33,19 @@ Java_net_butterflytv_rtmp_1client_RtmpClient_nativeOpen(JNIEnv* env, jobject thi
     RTMP *rtmp = (RTMP *) rtmpPointer;
    // rtmp = RTMP_Alloc();
     if (rtmp == NULL) {
-        return -1;
+        throwIllegalStateException(env, "RTMP open called without allocating rtmp object");
+        return RTMP_ERROR_IGNORED;
     }
 
+    rtmp->m_error = RTMP_SUCCESS;
     RTMP_Init(rtmp);
     rtmp->Link.receiveTimeoutInMs = receiveTimeoutInMs;
     rtmp->Link.sendTimeoutInMs = sendTimeoutInMs;
     int ret = RTMP_SetupURL(rtmp, url);
 
     if (!ret) {
         RTMP_Free(rtmp);
-        return -2;
+        return rtmp->m_error;
```

Comment:
It looks like `m_error` is being used in a pattern where it's set to `SUCCESS`, then some internal method is called that will change `m_error` on failure, and then if the method returns false the caller reads the value of `m_error` and returns that to its caller.

Is this pattern definitely necessary? Can the internal methods just return an `RTMPResult` directly? For example here would it be viable to do:

```
int result = RTMP_SetupURL(rtmp, url);
if (result != RTMP_SUCCESS) {
  RTMP_Free(rtmp);
  return result;
}
```

That kind of pattern seems a lot easier to follow, and is also consistent with the pattern now being used in the Java class. If we could get rid of `m_error` completely, that would seem like a great outcome!

Reply:
Done.